x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/EvalOperator.java

```diff
@@ -63,6 +63,17 @@ public interface ExpressionEvaluator extends Releasable {
         /** A Factory for creating ExpressionEvaluators. */
         interface Factory {
             ExpressionEvaluator get(DriverContext context);
+
+            /**
+             * {@code true} if it is safe and fast to evaluate this expression eagerly
+             * in {@link ExpressionEvaluator}s that need to be lazy, like {@code CASE}.
+             * This defaults to {@code false}, but expressions
+             * that evaluate quickly and can not produce warnings may override this to
+             * {@code true} to get a significant speed-up in {@code CASE}-like operations.
+             */
+            default boolean eagerEvalSafeInLazy() {
```

Comment:
another alternative is to be extract this as a marking interface that gets implemented by certain factories.